pySDC/projects/Second_orderSDC/stability_simulation.py

```diff
@@ -0,0 +1,287 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from pySDC.core.Errors import ProblemError
+from pySDC.core.Step import step
+
+from pySDC.projects.Second_orderSDC.plot_helper import set_fixed_plot_params
+from tabulate import tabulate
+class StabilityImplementation:
+    """
+    Routine to compute the stability domains of different configurations of SDC
+    """
+
+    def __init__(self, description, kappa_max=20, mu_max=20, Num_iter=(400, 400), cwd=''):
+        self.description = description
+        self.kappa_max = kappa_max
+        self.mu_max = mu_max
+        self.kappa_iter = Num_iter[0]
+        self.mu_iter = Num_iter[1]
+        self.lambda_kappa = np.linspace(0.0, self.kappa_max, self.kappa_iter)
+        self.lambda_mu = np.linspace(1e-10, self.mu_max, self.mu_iter)
+
+        self.K_iter = description['step_params']['maxiter']
+        self.num_nodes = description['sweeper_params']['num_nodes']
+        self.dt = description['level_params']['dt']
+        self.SDC, self.Ksdc, self.picard, self.Kpicard = self.stability_data()
+        self.cwd = cwd
+
+    def stability_data(self):
```

Comment:
This function is pretty long and difficult to understand.
Maybe it would make sense to split this function for the different types of iterations? Or to annotate it with some comments as to what you are doing?

Reply:
I wrote down some comments